common/models/user.js

```diff
@@ -870,6 +870,16 @@ module.exports = function(User) {
     next();
   });
 
+  User.observe('before save', function emptyPasswordUpdate(ctx, next) {
+    if ((ctx.instance && ctx.instance.password === '') || (ctx.data && ctx.data.password === '')) {
```

Comment:
One reason why I am opposing this new "before save" hook is inconsistency with existing password validations (e.g. max length). Even if this new hook fixed the issue for an empty password, it will most likely not fix it for other validations like max length. Additionally, it will make the code more difficult to maintain, because the validation logic will be spread across multiple places in the source code file.